Create ui-c8y-1023-14-24-when-no-data-exist-the-chart-is-still-rendered-properly.md (#4066)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1023-14-24-when-no-data-exist-the-chart-is-still-rendered-properly.md

---
date: ""

> title: Data point graph now renders properly when no data exists.

Reviewer: ```suggestion
title: Data point graph now renders properly when no data exists
```

product_area: Application enablement & solutions
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-65429
version: 1023.14.24
---

> Previously, selecting a time range with no measurement data caused the graph widget to render a completely empty area, without any message or placeholder indicating the absence of data. This has been fixed, and the graph now renders properly, providing a consistent display even when no data is available.

Reviewer: ```suggestion
Previously, when selecting a time range in the data point graph with no measurement data, the "Data point graph" widget rendered an empty area, without any indication that no data is available. This behavior has been changed. The "Data point graph" widget now provides a consistent display, showing the x-axis and y-axis even when no data is available.
```

Reviewer: @eniosultan Is this correct? Or what do you mean with "the graph now renders properly"?

Reviewer 2: @BeateRixen  It's no message, just before it was looking like it is broken, see image below:

before:

<img width="968" height="584" alt="image" src="https://github.com/user-attachments/assets/758f3979-7d1a-4594-868d-007105c0e725" />

now when there is no data we still display xAxis, yAxis the helper lines and everything...just without the data itself.

---
date: ""

> title: When no data exist, the chart is still rendered properly.

Reviewer 2: ```suggestion
title: Data point graph now renders properly when no data exists.
```

product_area: Application enablement & solutions
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-65429
version: 1023.14.24
---
# Backport

This will backport the following commits from `develop` to
`release/y2026`:
- [fix(Web SDK): [MTM-65429] When no data exist, the chart is still
rendered properly.
(#10519)](https://github.com/Cumulocity-IoT/cumulocity-ui/pull/10519)

<!--- Backport version: 9.5.1 -->

### Questions ?
Please refer to the [Backport tool
documentation](https://github.com/sorenlouv/backport)

[MTM-65429]:
https://cumulocity.atlassian.net/browse/MTM-65429?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Co-authored-by: Enio Sultan <92023325+eniosultan@users.noreply.github.com>

> Co-authored-by: Paweł Rynarzewski <92171763+pawel-rynarzewski-c8y@users.noreply.github.com>

Reviewer 2: ```suggestion
Previously, selecting a time range with no measurement data caused the graph widget to render a completely empty area, without any message or placeholder indicating the absence of data. This has been fixed, and the graph now renders properly, providing a consistent display even when no data is available.
```
